drivers/dac/ad5791/ad5791.c

```diff
@@ -409,3 +409,26 @@ int ad5791_spi_read_mask(struct ad5791_dev *dev,
 
 	return status;
 }
+
+/***************************************************************************//**
+ * @brief	Set Linearity error compensation based on the reference voltage span.
+ *
+ * @param	dev		- The device structure.
+ * @param	v_span	- voltage span to set the corresponding lin_comp value.
+ * @return	0 in case of success, negative error code otherwise.
+ */
+int ad5791_set_lin_comp(struct ad5791_dev *dev,
+			enum ad5791_lin_comp_select v_span)
+{
+	if (dev->act_device == ID_AD5781 || dev->act_device == ID_AD5791) {
```

Comment:
perhaps also check dev for NULL here, it could crash on the driver API user if they provide dev = NULL...

Reply:
this is an issue throughout the whole file, but those instances are unrelated to your change